js/web/lib/wasm/jsep/webgpu/ops/skip-layer-norm.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType, tensorTypeToWsglType} from '../../../wasm-common';
+import {TensorView} from '../../tensor';
+import {ShapeUtil} from '../../util';
+import {AttributeWithCacheKey, createAttributeWithCacheKey} from '../attribute-with-cache-key';
+import {ComputeContext, GpuDataType, ProgramInfo, ProgramInfoLoader, ProgramMetadata} from '../types';
+
+import {ShaderHelper} from './common';
+
+export interface SkipLayerNormAttributes extends AttributeWithCacheKey {
+  epsilon: number;
+}
+
+const validateInputs = (inputs: readonly TensorView[]): void => {
```

Comment:
since we are already registered using the ORT macros in C++, do we need to perform the checks in typesciprt?

Reply:
First of all please let me make it clear that the ONNX model, input/output connections are made using the names not inputs/output indices for the corresponding consuming or generating nodes.
For this model @dakenf  uploaded I verified that the SkipLayerNorm nodes are generating only two outputs, not 4. Hence there is no problem hitting the limit 8 buffers. Again, the consumer node does not look for a output from the produces at a specific index among its outputs. The index 3 here does not make the consumer look for the 4 output from the producer.
![image](https://github.com/microsoft/onnxruntime/assets/26722914/00132df4-0e71-4176-b7e2-bdd224126d3d)

Reply:
> First of all please let me make it clear that the ONNX model, input/output connections are made using the names not inputs/output indices for the corresponding consuming or generating nodes.

Then i believe it's not implemented for JSEP because if i remove this change, i get this
![image](https://github.com/microsoft/onnxruntime/assets/7466130/d136a983-3dcc-44e7-ab9f-6d5ceb71ad87)

```
failed to call OrtRun(). ERROR_CODE: 6, ERROR_MESSAGE: Non-zero status code returned while running SkipLayerNormalization node. Name:'SkipLayerNorm_AddBias_1' Status Message: C:/Users/me/projects/my/onnxruntime/include/onnxruntime\core/framework/op_kernel_context.h:42 const T *onnxruntime::OpKernelContext::Input(int) const [T = onnxruntime::Tensor] Missing Input: /down_blocks.0/attentions.0/transformer_blocks.0/Add_output_0
```

Reply:
I think I understand the problem. Let me check the best way to fix this.

Reply:
It throws this error now
```
The number of storage buffers (9) in the Compute stage exceeds the maximum per-stage limit (8).
 - While validating binding counts
 - While validating [BindGroupLayoutDescriptor]
 - While calling [Device].CreateComputePipeline([ComputePipelineDescriptor]).
```

Reply:
Did you pull the commit "Added output map"?

Reply:
> since we are already registered using the ORT macros in C++, do we need to perform the checks in typesciprt?
C++ macros don't check/match inputs shapes and dimensions. For example the code here checks hiddenSize and sequenceLength.